Basecamp_Project_Tests/WebApp.Tests/PointsControllerTest.cs

```diff
@@ -0,0 +1,168 @@
+            var mock = new Mock<IRepository<Point>>();
+            mock.Setup(repo => repo.Create(expectedTestPoint)).Returns(false);
+            var controller = new PointsController(mock.Object);
+            // Act
+            var result = controller.AddPoint(testPointRequest);
+            // Assert
+            result.Should().BeOfType<BadRequestObjectResult>();
+        }
+
+        [Fact]
+        public void UpdatePoint_requestIsValid_returnsOkResultWithUpdatedPoint()
+        {
+            // Arrange
+            UpdatePointRequest testUpdatePointRequest = new UpdatePointRequest(12, "Lab work №6 + extra task");
+            Point testPoint = new Point(5, 10, 1910205, "Object-orientate programing", DateTime.Now, "Lab work №6");
+            Point updatedTestPoint = new Point(testPoint.Id, testUpdatePointRequest.Mark, testPoint.StudentId, testPoint.Subject, DateTime.Now, testUpdatePointRequest.Task);
+            var mock = new Mock<IRepository<Point>>();
+            mock.Setup(repo => repo.Update(updatedTestPoint)).Returns(true);
+            mock.Setup(repo => repo.GetById(testPoint.Id)).Returns(testPoint);
+            var controller = new PointsController(mock.Object);
+            // Act
+            var result = controller.UpdatePoint(testPoint.Id, testUpdatePointRequest);
+            // Assert
+            result.Should().BeOfType<OkObjectResult>();
+            result.As<OkObjectResult>().Value.As<Point>().Should().BeEquivalentTo(updatedTestPoint);
+        }
+
+        [Fact]
+        public void UpdatePoint_specifiedPointNotExists_returnsNotFoundResult()
+        {
+            // Arrange
+            UpdatePointRequest testUpdatePointRequest = new UpdatePointRequest(12, "Lab work №6 + extra task");
+            Point testPoint = new Point(5, 10, 1910205, "Object-orientate programing", DateTime.Now, "Lab work №6");
+            Point updatedTestPoint = new Point(testPoint.Id, testUpdatePointRequest.Mark, testPoint.StudentId, testPoint.Subject, DateTime.Now, testUpdatePointRequest.Task);
+            var mock = new Mock<IRepository<Point>>();
+            mock.Setup(repo => repo.Update(updatedTestPoint)).Returns(false);
+            mock.Setup(repo => repo.GetById(testPoint.Id)).Returns(testPoint);
+            var controller = new PointsController(mock.Object);
+            // Act
+            var result = controller.UpdatePoint(testPoint.Id, testUpdatePointRequest);
+            // Assert
+            result.Should().BeOfType<NotFoundObjectResult>();
+        }
+
+        [Fact]
+        public void DeletePoint_specifiedPointExists_returnsOkResult()
+        {
+            // Arrange
+            var mock = new Mock<IRepository<Point>>();
+            mock.Setup(repo => repo.Delete(123)).Returns(true);
+            var controller = new PointsController(mock.Object);
+            // Act
+            var result = controller.DeletePoint(123);
```

Comment:
Please cover the repository calls with params. For ex.: mock.Verify(repo => repo.Delete(It.Is<int>(123)), Times.Once);